content/blog/2019/09/2019-09-16-outreachy-audit-log-release.adoc

```diff
@@ -0,0 +1,49 @@
+This plugin is the first major project completed related to Outreachy, and I'd like to give a brief overview of the functionality that was developed for this release.
+The primary goal of this plugin is to introduce an audit trail of various Jenkins events using structured logging and related audit logging standards.
+Initially, this plugin covers audit events related to core Jenkins concepts like user accounts, jobs, builds, nodes, and credentials usage.
+More specifically, this tracks:
+
+* User login and logout events
+* Credentials usage
+* User creation (when using the Jenkins user database as a security realm)
+* User password updates (ditto)
+* Starts and ends of builds
+* Creation/modification/deletion/copying of items (which correspond to projects, pipelines, folders, etc.)
+* Creation/modification/deletion of nodes.
+
+This plugin defines and exports standardized log event classes and schemas corresponding to these events.
+Other plugins can add `audit-log` as a dependency to define their own audit events using https://logging.apache.org/log4j-audit/latest/[Apache Log4j Audit] and its catalog editor; then they can use the Maven plugin for generating the audit event classes for use in the plugin.
+
+The other major feature of this plugin is configuring where to output these audit logs.
+By default, audit logs will be written in HTML files (rotated once per day) to `$JENKINS_HOME/logs/html/audit.html` which are viewable through the "Audit Logs" root action link.
+In the system settings, a section for audit logging is added where the main audit log output can be configured.
+This can initially be configured to output via either a JSON log file in `$JENKINS_HOME/logs/audit.log` by default or to a syslog server using RFC5424 encoding.
+
+Overall, this experience has been rather interesting.
+Besides having an opportunity to mentor new contributors, Outreachy has helped open my eyes to the struggles that developers from around the world are dealing with which can be improved upon to help expand our communities.
+For example, many countries do not have reliable internet or electricity, so the use of synchronous videoconferencing and other heavyweight, synchronous processes common to more corporate-style development are inadequate in this international context.
+This doesn't even begin to account for the difference in timezones which is not always an issue, though both problems are addressable by using asynchronous communication methods like chat and email.
+This notion of asynchronous communication is an important aspect of the Apache Way, for example, which emphasises processes that allow for vendor neutral communities to form and thrive around a project.
+
+This mentoring project was valuable to myself as well.
+This audit log project was not led by any project managers, so this gave me an opportunity to develop my own skills in that area since I'm normally programming myself.
```

Comment:
The last bit of this sentence reads a little weird. I understand what it's trying to say, but "programming myself" sounds a little bit like you're a bot. Maybe a little rework to say something about programming being your usual role or something.